docs/model_developers/release.rst

```diff
@@ -36,42 +36,59 @@ When done, first use git to add files:
 
     git add *
 
-To commit, use the module tool. This will create a new dev release.
+Create a new dev release:
 
 .. code-block:: bash
 
-    inmanta module commit --patch -m "Fixed small bug"
+    inmanta module release --dev --patch -m "Fixed small bug"
 
-This command will set the version to the next dev version (`+0.0.1dev`) and add a timestamp.
+This command will set the version to the next dev version, e.g. ``+0.0.1dev`` here for a patch increment.
 
 The module tool supports semantic versioning.
-Use one of ``--major``, ``--minor`` or ``--patch`` to update version numbers: <major>.<minor>.<patch>
+Use one of ``--major``, ``--minor`` or ``--patch`` to update version numbers: ``<major>.<minor>.<patch>``
 
 For the dev releases, no tags are created.
+Once the dev version is ready for release, perform a proper release by following
+the steps in the `Release Versions`_ section below.
 
 Release Versions
 ----------------
 
-To make an actual release (without `.dev` at the end):
+To perform an actual non-dev release:
+
+First, merge your changes from your feature branch (e.g. ``mywork`` in the previous example) onto the branch dedicated to
+releases (e.g. master) and then do the actual release:
```

Comment:
I have to say I don't really see the relevance of this addition, and the code block below. We're not in the context of a pull request here, we're releasing a dev version that has presumably lived on master for a while already.